src/storage/migrations.js

```diff
@@ -348,3 +340,18 @@ export const migrations: {| [string]: (GlobalState) => GlobalState |} = {
 
   // TIP: When adding a migration, consider just using `dropCache`.
 };
+
+type PartialState = $ReadOnly<$Rest<GlobalState, { ... }>>;
+
+/**
+ * Migrations for data persisted by previous versions of the app.
+ *
+ * These are run by `redux-persist-migrate` when the previously persisted
+ * state is loaded ("rehydrated") by `redux-persist`; they transform that
+ * state object before it's applied to our live state.  The state includes
+ * a version number to track which migrations are already reflected in it,
+ * so that each only has to be run once.
+ */
+/* $FlowFixMe[incompatible-type] This is where we pretend that the storeKeys
+     are all present. */
```

Comment:
> Also, is the `$FlowFixMe` here accurate about the "outer" lie?

Ah, if you mean this one:
```js
/* $FlowFixMe[incompatible-type] This is where we pretend that the storeKeys
     are all present. */
export const migrations: {| [string]: (PartialState) => PartialState |} = migrationsInner;
```

then that applies to where we take the inner type and cast it to the outer type -- so it's about the gap between the lie told inside the module and the smaller one told on the outside.

The "outer" lie is that we pretend the object passed to these migrations is a `PartialState`. That's untrue because if it has, for example, an `accounts` property, that property may not have the type that `GlobalState` (and hence `PartialState`) expect for the `accounts` property. But no code inside this module is going to be aware that's a lie, because the `migrations` we export already expects `PartialState` for the migrations' arguments.

The "inner" lie is that we pretend the object passed to these migrations is a `LessPartialState`. That's untrue for the same reason as the first one; plus that `LessPartialState` means that `accounts` and the other `storeKeys` are actually all present, while in reality some keys may not be present. So that latter, additional, lie is the one that this `$FlowFixMe` has to cover.